pkg/operator/ceph/cluster/controller.go

```diff
@@ -805,13 +799,87 @@ func (c *ClusterController) handleDelete(cluster *cephv1.CephCluster, retryInter
 		}
 
 		logger.Infof("waiting for volume attachments in cluster %s to be cleaned up. Retrying in %s.",
-			cluster.Namespace, retryInterval)
+			cluster.Namespace, retryInterval.String())
 		<-time.After(retryInterval)
 	}
+	return nil
+}
+
+func (c *ClusterController) checkPVPresentInCluster(drivers []string, clusterID string) (bool, error) {
+	pv, err := c.context.Clientset.CoreV1().PersistentVolumes().List(metav1.ListOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	for _, p := range pv.Items {
+		if p.Spec.CSI == nil {
+			logger.Errorf("Spec.CSI is nil for PV %s", p.Name)
+			continue
+		}
+		if p.Spec.CSI.VolumeAttributes["clusterID"] == clusterID {
+			//check PV is created by drivers deployed by rook
+			for _, d := range drivers {
+				if d == p.Spec.CSI.Driver {
+					return true, nil
+				}
+			}
+
+		}
+	}
+	return false, nil
+}
+
+func (c *ClusterController) waitForCSIVolumeCleanup(cluster *cephv1.CephCluster, retryInterval time.Duration) error {
+	retryCount := 0
+	drivers := []string{csi.CephFSDriverName, csi.RBDDriverName}
+	for {
+		logger.Infof("checking any PVC created by drivers %s and %s with clusterID %s", csi.CephFSDriverName, csi.RBDDriverName, cluster.Namespace)
+		// check any PV is created in this cluster
+		attachmentsExist, err := c.checkPVPresentInCluster(drivers, cluster.Namespace)
+		if err != nil {
+			err = fmt.Errorf("failed to list PersistentVolumes %+v", err)
```

Comment:
```suggestion
			err = fmt.Errorf("failed to list PersistentVolumes. %+v", err)
```

The punctuation is important so we know when the error message begins.